aws/sign.go

```diff
@@ -15,28 +15,35 @@ import (
 	"time"
 )
 
+// SignerFunc represents a function that can sign a http.Request with the given Auth details.
+type SignerFunc func(*http.Request, Auth, string) error
+
 type V2Signer struct {
-	auth    Auth
-	service ServiceInfo
-	host    string
+	auth     Auth
+	endpoint string
 }
 
 var b64 = base64.StdEncoding
 
-func NewV2Signer(auth Auth, service ServiceInfo) (*V2Signer, error) {
-	u, err := url.Parse(service.Endpoint)
-	if err != nil {
-		return nil, err
-	}
-	return &V2Signer{auth: auth, service: service, host: u.Host}, nil
+func NewV2Signer(auth Auth, endpoint string) (*V2Signer, error) {
+	return &V2Signer{auth: auth, endpoint: endpoint}, nil
```

Comment:
Might be missing something here but can this fun ever return a non nil error? If not is there a point to keep the error in the return.